### PR TITLE
Add "code" (shell) linter and format checker

### LINKED ISSUE #13: Project maintenance: shfmt
Add a formatter for the scripts

### LINKED ISSUE #14: Project maintenance: shellcheck
Add a script linter

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
This uses:

- [ShellCheck](https://www.shellcheck.net/)
- [shfmt](https://github.com/mvdan/sh)

behind the scenes.

After #1 is merged I don't mind either relaxing the rules or adapting the code. I've found, over use, that at least ShellCheck is very good at catching error-prone patterns and even helping make it more readable.

Closes #13.
Closes #14.

### CODE REVIEW

**Comment by paulo-ferraz-oliveira** [comment]
Now that #1 is out this is rebased on top of the default branch.

**Comment by paulo-ferraz-oliveira** [comment]
Branch is updated and rebased on top of the default branch. Lemme know what you want relaxed, commented, removed, etc.

File: scripts/build_otp_macos.bash [hunk 1]
```diff
@@ -118,7 +118,7 @@ test_otp() {
     {ok, _} = application:ensure_all_started(crypto), io:format("crypto ok~n"),
     halt().'
 
-  if dyld_info ${OTP_DIR}/lib/crypto-*/priv/lib/crypto.so | tail -n +2 | grep -q openssl; then
+  if dyld_info "${OTP_DIR}/lib/crypto-*/priv/lib/crypto.so" | tail -n +2 | grep -q openssl; then
```
Comment (starbelly) [inline scripts/build_otp_macos.bash:121]: That's handy!

**Review by starbelly — APPROVED** [review]
LGTM

**Comment by paulo-ferraz-oliveira** [comment]
As with the other one, I'm merging and it should be easy to revert if it causes issues in the default branch.

File: .github/workflows/check.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+---
+"on":
```
Comment (wojtekmach) [inline .github/workflows/check.yaml:2]: > `"on"`

I'm not a fan of this particular change but I get it and it's obviously not a deal breaker. Thank you for the PR!

Comment (paulo-ferraz-oliveira) [reply .github/workflows/check.yaml:2]: We can revert it. It's an option for the linter. I'll do it.

**Pull request merged**